scrapi/harvesters/usgs.py

```diff
@@ -0,0 +1,124 @@
+        author_d['additionalName'] = name.middle
+        author_d['givenName'] = name.first
+        author_d['familyName'] = name.last
+    return all_processed_authors
+
+
+class USGSHarvester(JSONHarvester):
+    short_name = 'usgs'
+    long_name = 'United States Geological Survey'
+    url = 'https://pubs.er.usgs.gov/'
+    DEFAULT_ENCODING = 'UTF-8'
+
+    URL = 'https://pubs.er.usgs.gov/pubs-services/publication?'
+
+    schema = {
+        'title': '/title',
+        'description': '/docAbstract',
+        'providerUpdatedDateTime': ('/lastModifiedDate', datetime_formatter),
+        'uris': {
+            'canonicalUri': ('/id', 'https://pubs.er.usgs.gov/publication/{}'.format),
+            'providerUris': [('/id', 'https://pubs.er.usgs.gov/publication/{}'.format)],
+            'descriptorUris': [('/doi', 'https://dx.doi.org/{}'.format)]
+        },
+
+        'contributors': ('/contributors/authors', process_contributors),
+        'otherProperties': build_properties(
+            ('serviceID', ('/id', str)),
+            ('definedType', '/defined_type'),
+            ('type', '/type'),
+            ('links', '/links'),
+            ('publisher', '/publisher'),
+            ('publishedDate', '/displayToPublicDate'),
+            ('publicationYear', '/publicationYear'),
+            ('issue', '/issue'),
+            ('volume', '/volume'),
+            ('language', '/language'),
+            ('indexId', '/indexId'),
+            ('publicationSubtype', '/publicationSubtype'),
+            ('startPage', '/startPage'),
+            ('endPage', '/endPage'),
+            ('onlineOnly', '/onlineOnly'),
+            ('additionalOnlineFiles', '/additionalOnlineFiles'),
+            ('country', '/country'),
+            ('state', '/state'),
+            ('ipdsId', '/ipdsId'),
+            ('links', '/links'),
+            ('doi', '/doi'),
+            ('contributors', '/contributors'),
+            ('otherGeospatial', '/otherGeospatial'),
+            ('geographicExtents', '/geographicExtents'),
+
+        )
+    }
+
+    def harvest(self, start_date=None, end_date=None):
+
+        # This API does not support date ranges
+        start_date = start_date or date.today() - timedelta(settings.DAYS_BACK)
+        
+        # days_back = the number of days between start_date and now, defaulting to settings.DAYS_BACK
+        days_back = settings.DAYS_BACK
+        search_url = '{0}mod_x_days={1}'.format(
+            self.URL,
+            days_back
+        )
+
+        record_list = []
+        for record in self.get_records(search_url):
+            doc_id = record['id']
+
+            record_list.append(
+                RawDocument(
+                    {
+                        'doc': json.dumps(record),
+                        'source': self.short_name,
+                        'docID': six.text_type(doc_id),
+                        'filetype': 'json'
+                    }
+                )
+            )
+
+        return record_list
```

Comment:
This whole block can actually just be done in a list comprehension, which is how we generally prefer things to work.
```python
return [
    RawDocument({
        'doc': json.dumps(record),
        'source': self.short_name,
        'docID': six.text_type(record['id']),
        'filetype': 'json'
    }) for record in self.get_records(search_url)
]
```